app/controllers/users_controller.rb

```diff
@@ -254,6 +261,17 @@ def handle_settings_tab
 
   private
 
+  def set_or_sign_in_user
+    @user = current_user
+
+    if @user
+      authorize @user
+    else
+      flash[:alert] = "You must be logged in to proceed with account deletion."
+      redirect_to new_user_session_path
```

Comment:
I think this should be `sign_up_path` instead, right? At least it's the one I get on `http://localhost:3000/rails/info/routes` to match the `/enter` path.

Also a small heads up: If you're moving this block into the action don't forget the `return` statement after the redirect, because the action won't stop executing after a redirect 🙂

Reply:
Agree!! This is what I was thinking of when I left my earlier comment about redirects :) if you have more than one, you'll have to return out of them explicitly